src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -15,13 +15,33 @@ namespace System.Device.Gpio.Drivers
     public class SysFsDriver : UnixDriver
     {
         private const string GpioBasePath = "/sys/class/gpio";
+        private const string GpioChip = "gpiochip";
+        private const string GpioLabel = "/label";
+        private const string GpioContoller = "pinctrl";
+        private const string GpioOffsetBase = "/base";
         private int _pollFileDescriptor = -1;
         private Thread _eventDetectionThread;
         private int _pinsToDetectEventsCount;
         private readonly static CancellationTokenSource s_eventThreadCancellationTokenSource = new CancellationTokenSource();
         private readonly List<int> _exportedPins = new List<int>();
         private readonly Dictionary<int, UnixDriverDevicePin> _devicePins = new Dictionary<int, UnixDriverDevicePin>();
         private readonly int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
+        private int _pinOffset = 0;
+
+        public SysFsDriver() {
+            IEnumerable<string> fileNames = Directory.EnumerateFileSystemEntries(GpioBasePath);
+            foreach (string name in fileNames)
+            {
+                if (name.Contains(GpioChip))
+                {
+                    if (File.ReadAllText($"{name}{GpioLabel}").StartsWith(GpioContoller))
```

Comment:
>File [](start = 24, length = 4)

should we probe for the files existence before reading them?

Reply:
I think it's unexpected for the file to not exist so I think error would be fine if that happened

Reply:
Yes but I believe we should throw describable exception and not just having the caller see IO Exception for a file not exist while not having any idea why getting that.

Reply:
With several Linux versions (Rasbian, Debian) on Raspberry Pi i have tested, these path existed, but cannot guarantee the same for all linux versions. So agree with checking file existance, but don't agree with throwing, I think we should continue with default `0` offset in that case

Reply:
ok, just write a comment to clarify that. Thanks.

---
In reply to: [278330737](https://github.com/dotnet/iot/pull/386#discussion_r278330737) [](ancestors = 278330737)